tests/test_project.py

```diff
@@ -1480,85 +1480,68 @@ def test_import_own_project(self):
-                            raise pytest.skip('requires use_pandas')
-                        self.project._repr_html_()
-
-        with subtests.test(of='JobsCursor'):
-            for filter_ in (None, ):
-                with subtests.test(filter=filter_):
-                    with subtests.test(type='str'):
-                        str(self.project.find_jobs(filter_))
-                    with subtests.test(type='repr'):
-                        q = self.project.find_jobs(filter_)
-                        assert eval(repr(q)) == q
-                    with subtests.test(type='html'):
-                        for use_pandas in (True, False):
-                            type(self.project)._use_pandas_for_html_repr = use_pandas
-                            with subtests.test(use_pandas=use_pandas):
-                                if use_pandas and not PANDAS:
-                                    raise pytest.skip('requires use_pandas')
-                                self.project.find_jobs(filter_)._repr_html_()
-
-        with subtests.test(of='Schema'):
-            schema = self.project.detect_schema()
-            with subtests.test(type='str'):
-                str(schema)
-            with subtests.test(type='repr'):
-                repr(schema)
-            with subtests.test(type='html'):
-                schema._repr_html_()
-
-    def test_repr_no_jobs(self, subtests):
-        self.call_repr_methods(subtests)
-
-    def test_repr_few_jobs_homogeneous(self, subtests):
-        # Many jobs with many different state points
-        for i in range(self.num_few_jobs):
-            self.project.open_job(
-                {'{}_{}'.format(i, j): v
-                 for j, v in enumerate(self.valid_sp_values)}).init()
-        self.call_repr_methods(subtests)
-
-    def test_repr_many_jobs_homogeneous(self, subtests):
-        # Many jobs with many different state points
-        for i in range(self.num_many_jobs):
-            self.project.open_job(
-                {'{}_{}'.format(i, j): v
-                 for j, v in enumerate(self.valid_sp_values)}).init()
-        self.call_repr_methods(subtests)
-
-    def test_repr_few_jobs_heterogeneous(self, subtests):
-        # Many jobs with many different state points
-        for i in range(self.num_few_jobs):
-            for v in self.valid_sp_values:
-                self.project.open_job(dict(a=v)).init()
-        self.call_repr_methods(subtests)
-
-    def test_repr_many_jobs_heterogeneous(self, subtests):
-        # Many jobs with many different state points
-        for i in range(self.num_many_jobs):
-            for v in self.valid_sp_values:
-                self.project.open_job(dict(a=v)).init()
-        self.call_repr_methods(subtests)
+    @pytest.mark.parametrize("project_generator,num_jobs", project_repr_generators)
+    def test_project_repr_methods(self, project_generator, num_jobs):
+        project_generator(self.project, num_jobs)
+        str(self.project)
```

Comment:
Should we do a super simple sanity check here, something like `assert len(self.project) > 0` or `assert 'project' in str(self.project)` ?

Reply:
Good suggestion! I added `assert len(str(self.project)) > 0` and some related tests. The other case is already handled twice by `TestProject.test_get_id` and `TestProject.test_property_id`.